src/DynamoCore/Models/DynamoModelCommands.cs

```diff
@@ -296,11 +296,16 @@ void BeginConnection(Guid nodeId, int portIndex, PortType portType)
 
         void EndConnection(Guid nodeId, int portIndex, PortType portType)
         {
-            bool isInPort = portType == PortType.Input;
+            // Check if the node from which the connector starts is valid and has not been deleted
+            if (activeStartPort.Owner == null) return;
+
+            var startNode = CurrentWorkspace.GetModelInternal(activeStartPort.Owner.GUID);
+            if (startNode == null) return;
 
             var node = CurrentWorkspace.GetModelInternal(nodeId) as NodeModel;
-            if (node == null)
-                return;
+            if (node == null) return;
+
+            bool isInPort = portType == PortType.Input;
```

Comment:
As discussed... let's handle `DeleteModelCommand` and get the `StateMachine` out of its current state (i.e. connector creation state) and properly restore what it needs to restore. Thanks!